test/slow/api_spec_slow.js

```diff
@@ -197,15 +197,43 @@ describe(`electron-forge API (with installer=${installer.substr(12)})`, () => {
       const genericTargets = fs.readdirSync(path.resolve(__dirname, '../../src/makers/generic')).map(file => path.parse(file).name);
 
       [].concat(targets).concat(genericTargets).forEach((target) => {
-        describe(`make (with target=${target})`, () => {
-          before(async () => {
-            const packageJSON = await readPackageJSON(dir);
-            packageJSON.config.forge.make_targets[process.platform] = [target];
-            await fs.writeFile(path.resolve(dir, 'package.json'), JSON.stringify(packageJSON));
-          });
-
-          it('successfully makes with default config', async () => {
-            await forge.make({ dir, skipPackage: true });
+        [{
+          format: 'missing',
+          value: undefined,
+          expectFailure: true,
+        }, {
+          format: 'simple-string',
+          value: 'First Last',
+        }, {
+          format: 'full-string',
+          value: 'First Last <first.last@example.com> (http://www.example.com/)',
+        }, {
+          format: 'object',
+          value: {
+            name: 'First Last',
+            email: 'first.last@example.com',
+            url: 'http://www.example.com/',
+          },
+        }].forEach((author) => {
```

Comment:
The reasons I didn't really like that idea were that it can't test the exceptional cases and can't ensure that `electron-windows-store` actually gets called with valid options (used to something more like [`td.replace`](https://github.com/testdouble/testdouble.js/blob/master/docs/7-replacing-dependencies.md)). Plus I suspect that this functionality is unlikely to be useful in other contexts of this application. Nevertheless, a reasonable level of test coverage is still possible, and these tests are indeed much faster.

Also, are you OK with using the [`parse-author`](https://github.com/jonschlinkert/parse-author) module (also MIT license)? In a previous revision I had suggested just using the author string in its entirety, but that leads to the perhaps surprising result that if you have author information as a string with name, email, and url then convert it to object format you'll see a different result. For predictability I think we'd always want to either just extract the name or generate a full author string. The commit I just pushed (`33878c0`) assumes the former.